### PR TITLE
feat(split-button): DLT-1817 new component

### PR DESCRIPTION
**Pull request by francisrupert**
# Design and HTML/CSS for Split Button
3d293cb3-111a-4860-a945-e8b1b8d848ab_text

## :hammer_and_wrench: Type Of Change

These types will increment the version number on release:

- [ ] Fix
- [x] Feature
- [ ] Performance Improvement
- [ ] Refactor

## :book: Jira Ticket

* https://dialpad.atlassian.net/browse/DLT-1817
* https://dialpad.atlassian.net/browse/DLT-1818

## :book: Description

Design and HTML/CSS creation of new Split Button component. 

## :pencil: Checklist

For all PRs:

- [x] I have ensured no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).
- [x] I have reviewed my changes.
- [ ] I have added all relevant documentation.
- [x] I have considered the performance impact of my change.

For all CSS changes:

- [x] I have used design tokens whenever possible.
- [x] I have considered how this change will behave on different screen sizes.
- [x] I have visually validated my change in light and dark mode.
- [x] I have used gap or flexbox properties for layout instead of margin whenever possible.

If new component:

- I am exporting any new components or constants:
  - [ ] from the index.js in the component directory.
  - [ ] from the index.js in the root (either `packages/dialtone-vue2` or `packages/dialtone-vue3`).
- [x] I have added the styles for the new component to the `packages/dialtone-css` package.
- [ ] I have created a page for the new component on the documentation site in `apps/dialtone-documentation`.
- [ ] I have added the new component to `common/components_list.cjs`
- [ ] I have created a component story in storybook
- [ ] I have created story / stories for any relevant component variants in storybook
- [ ] I have created a docs page for the component in storybook.
- [ ] I have checked that changing all props/slots via the UI in storybook works as expected.

## :crystal_ball: Next Steps

Once the Design and its HTML/CSS is approved, the Vue component will be built in https://dialpad.atlassian.net/browse/DLT-1819

## :camera: Screenshots / GIFs

image

### CODE REVIEW

**Comment by francisrupert** [comment]
**NOTE:** feedback request is purely about the HTML structure, its CSS, and how I've used `<dt-button>` in the service of making it a new Vue component.

View it on https://dialtone.dialpad.com/deploy-previews/pr-373/components/split-button.html

**Pick it apart. Try to break it. Stress-test it. Find the holes.**

File: packages/dialtone-css/lib/build/less/components/button.less [hunk 1]
```diff
@@ -534,9 +534,115 @@
...
+            }
+        }
+
+        // bleed divider line to absolute top and bottom of the button
+        &.d-btn:disabled::before,
+        &.d-btn--outlined::before,
+        &.d-btn--primary::before {
+            top: var(--dt-size-100-negative);
+            bottom: var(--dt-size-100-negative);
+        }
```
Comment (braddialpad) [inline packages/dialtone-css/lib/build/less/components/button.less]: Seems like when toggling disabled in the example the screen "jumps" a bit implying it's a different size than it was. Is this intended?

File: apps/dialtone-documentation/docs/components/button.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ---
 title: Button
 description: A button is an UI element which signals key actions to take an action throughout an app. It is important a button is identifiable, consistent, communicates its actions clearly, and is appropriately sized to its action.
-status: ready
+status: Beta
```
Comment (braddialpad) [inline apps/dialtone-documentation/docs/components/button.md]: button is in beta?

File: packages/dialtone-css/lib/build/less/components/button.less [hunk 3]
```diff
@@ -534,9 +534,115 @@
...
 //  $   BUTTON GROUP
 //  ----------------------------------------------------------------------------
+
+.asdfasdfqwerqwer {
+    position: fixed;
+    z-index: var(--zi-notification);
+    inset: var(--dt-size-700) var(--dt-size-0) var(--dt-size-0);
+    overflow: auto;
+    background-color: var(--dt-color-surface-primary);
+}
```
Comment (braddialpad) [inline packages/dialtone-css/lib/build/less/components/button.less]: just for the temporary example I assume? haha

**Comment by braddialpad** [comment]
changing width looks good after your recent update 👍 

Screenshot 2024-06-28 at 10 44 56 AM

File: apps/dialtone-documentation/docs/components/button.md (hunk 2 above)
Comment (francisrupert) [reply apps/dialtone-documentation/docs/components/button.md]: Reverted it. That's when I was using it as a template for split button md.

File: packages/dialtone-css/lib/build/less/components/button.less (hunk 1 above)
Comment (francisrupert) [reply packages/dialtone-css/lib/build/less/components/button.less]: That's actually an existing issue for DtButton. Disabled buttons don't have a consistent use of borders as all other variants.

I'll revisit that separately some time, but also we probably will redo the disabled state of all variants so they don't all look the same.

File: packages/dialtone-css/lib/build/less/components/button.less (hunk 3 above)
Comment (francisrupert) [reply packages/dialtone-css/lib/build/less/components/button.less]: Just a convenient way to show the grid over 100% of viewport.

File: packages/dialtone-css/lib/build/less/components/button.less (hunk 1 above)
Comment (braddialpad) [reply packages/dialtone-css/lib/build/less/components/button.less]: ah okay np then

**Review by braddialpad — APPROVED** [review]
Looks good, I guess we won't merge since we'll just be building off of this branch.

**Comment by francisrupert** [comment]
> ...building off of this branch.

Coolness. I'll manually close [DLT-1818](https://dialpad.atlassian.net/browse/DLT-1818).

[DLT-1818]: https://dialpad.atlassian.net/browse/DLT-1818?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Pull request merged**